Controllers/LoginScreen.swift

```diff
@@ -12,86 +12,92 @@ import CoreData
     }
+    if !staticIndex.firstLaunch {
+      authentificationService = AuthentificationService(username: "", password: "")
+      self.authentifyTheUser()
+      staticIndex.firstLaunch = true
+    }
+  }
+
+  // MARK: Actions
+
+  func getLoggedStatus() -> Bool {
+    let context = appDelegate.persistentContainer.viewContext
+    let request = NSFetchRequest<NSFetchRequestResult>(entityName: "Users")
+
+    request.returnsObjectsAsFaults = false
+    var usersInfo = [NSManagedObject]()
 
-    func checkLoggedStatus(loggedStatus: Bool, token: String?, error: OAuth2Error?) {
-        if (!loggedStatus) || (loggedStatus && token == nil) {
-            if error?.description == "The Internet connection appears to be offline." {
-                self.performSegue(withIdentifier: "SegueNoInternetOnFirstConnection", sender: self)
-            } else {
-                let alert = UIAlertController(title: "Veuillez réessayer", message: "Identifiant inconnu ou mot de passe incorrect.", preferredStyle: .alert)
-
-                alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
-                self.present(alert, animated: true)
-            }
-        } else if loggedStatus && token != nil {
-            self.performSegue(withIdentifier: "SegueFromLogScreenToConnected", sender: self)
-        }
+    do {
+      usersInfo = try context.fetch(request) as! [NSManagedObject]
+    } catch {
+      print("Fetch failed")
     }
+    let loggedStatus: Bool = (usersInfo[(usersInfo.count) - 1].value(forKey: "loggedStatus") != nil)
 
-    func authentifyTheUser() {
-        if !entityIsEmpty(entity: "Users") {
-            let _ = authentificationService?.authorize(presenting: self)
-            var token = self.authentificationService?.oauth2.accessToken
-            var loggedStatus = self.getLoggedStatus()
-
-            authentificationService?.oauth2.afterAuthorizeOrFail = { authParameters, error in
-                token = self.authentificationService?.oauth2.accessToken
-                loggedStatus = self.getLoggedStatus()
-                self.checkLoggedStatus(loggedStatus: loggedStatus, token: token, error: error)
-            }
-            if loggedStatus && token != nil {
-                self.performSegue(withIdentifier: "SegueFromLogScreenToConnected", sender: self)
-            }
-        } else {
-            let _ = authentificationService?.authorize(presenting: self)
-            authentifyTheUser()
-        }
+    return loggedStatus
+  }
+
+  func checkLoggedStatus(loggedStatus: Bool, token: String?, error: OAuth2Error?) {
+    if (!loggedStatus) || (loggedStatus && token == nil) {
+      if error?.description == "The Internet connection appears to be offline." {
+        performSegue(withIdentifier: "SegueNoInternetOnFirstConnection", sender: self)
+      } else {
+        let alert = UIAlertController(title: "Veuillez réessayer", message: "Identifiant inconnu ou mot de passe incorrect.", preferredStyle: .alert)
```

Comment:
This should be I18n-ed.
Either we do it now or we add a `#TODO: I18n` next to it.